react/src/pages/analysis/Analysis.tsx

```diff
@@ -136,7 +141,7 @@ export default function Analysis() {
                 <AnalysisInfoDialog
                     open={detail}
                     analysis={activeRow}
-                    onClose={() => setDetail(false)}
+                    onClose={() => {setDetail(false); if(analysisID){history.push("/dashboard")}}}
```

Comment:
I think [`history.goBack()`](https://reactrouter.com/web/api/history) would be better so we can have multiple points of entry to this view.